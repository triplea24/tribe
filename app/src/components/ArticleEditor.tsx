import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { Theme } from "@material-ui/core";
import Paper from "@material-ui/core/Paper";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import AttachmentIcon from "@material-ui/icons/Attachment";
import Chip from "@material-ui/core/Chip";
import { Editor } from "react-draft-wysiwyg";
import { EditorState, ContentState } from "draft-js";
import { connect } from "react-redux";
import "react-draft-wysiwyg/dist/react-draft-wysiwyg.css";

import { changeBodyContent, changeTitle } from "../actions";

interface Props {
  classes?: any;
  changeBodyContent?: any;
  changeTitle?: any;
  body: string;
  title: string;
}
interface State {
  editorState: EditorState;
}

const chips = [
  { key: "0", label: "OC" },
  { key: "1", label: "SPOILER" },
  { key: "2", label: "NSFW" }
];

class ArticleEditor extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      editorState: EditorState.createWithContent(
        ContentState.createFromText(props.body)
      )
    };
  }
  onEditorStateChange = (editorState: EditorState) => {
    this.setState(
      {
        editorState
      },
      () => {
        this.props.changeBodyContent(
          // TODO: Not plaintext here!
          editorState.getCurrentContent().getPlainText()
        );
      }
    );
  };

  handleChangeTitle = ({ target: { value: body } }: any) => {
    this.props.changeTitle(body);
  };

  render() {
    const { classes } = this.props;
    const bodyPlaceholder = `Text ... (optional)`;
    const titlePlaceholder = "Title";
    const Chips = chips.map(({ label, key }) => (
      <Grid item xs={1} key={key}>
        <Chip className={classes.tag} label={label} />
      </Grid>
    ));
    return (
      <Paper className={classes.root}>
        <Grid container spacing={8}>
          <Grid item lg={12}>
            <TextField
              className={classes.title}
              multiline={true}
              fullWidth={true}
              placeholder={titlePlaceholder}
              value={this.props.title}
              onChange={this.handleChangeTitle}
            />
          </Grid>
          <Grid item lg={12}>
            <Editor
              placeholder={bodyPlaceholder}
              editorState={this.state.editorState}
              toolbarClassName="toolbarClassName"
              wrapperClassName="wrapperClassName"
              editorClassName="editorClassName"
              onEditorStateChange={this.onEditorStateChange}
            />
          </Grid>
          {Chips}
          <Grid item lg={7} />
          <Grid item lg={1}>
            <Button variant="outlined" className={classes.fileButton}>
              <AttachmentIcon />
              File
            </Button>
          </Grid>
          <Grid item lg={1}>
            <Button className={classes.button}>Post</Button>
          </Grid>
        </Grid>
      </Paper>
    );
  }
}

const styles = ({ spacing }: Theme) => ({
  root: {
    flexGrow: 1,
    padding: 18
  },
  body: {},
  button: {
    background: "#34aa44",
    color: "#ffffff",
    paddingLeft: 24,
    paddingRight: 24,
    paddignTop: 8,
    paddingBottom: 8,
    size: 14
  },
  fileButton: {
    paddingLeft: 15,
    paddingRight: 15,
    paddignTop: 8,
    paddingBottom: 8,
    size: 14
  },
  tag: {}
});

const mapStateToProps = ({ bodyContent: body = "", title }: any) => ({
  title,
  body
});

const withMaterialUI = withStyles(styles);
const withRedux = connect(
  mapStateToProps,
  { changeBodyContent, changeTitle }
);

export default withRedux(withMaterialUI(ArticleEditor));
