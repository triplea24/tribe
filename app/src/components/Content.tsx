import React from "react";
import { Theme } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import Avatar from "@material-ui/core/Avatar";
import Typography from "@material-ui/core/Typography";

const styles = (theme: Theme) => ({
  root: {
    width: "100%",
    backgroundColor: theme.palette.background.paper
  },
  inline: {
    display: "inline"
  }
});

interface Props {
  classes?: any;
  avatar: string;
  avatarAlt: string;
  type: string;
  title?: string;
  body: string;
}

class Content extends React.Component<Props> {
  render() {
    const { classes } = this.props;
    return (
      <ListItem alignItems="flex-start">
        <ListItemAvatar>
          <Avatar alt={this.props.avatarAlt} src={this.props.avatar} />
        </ListItemAvatar>
        <ListItemText
          primary={this.props.type === "article" && this.props.title}
          secondary={
            <Typography
              component="span"
              className={classes.inline}
              color="textPrimary"
            >
              {this.props.body}
            </Typography>
          }
        />
      </ListItem>
    );
  }
}

export default withStyles(styles)(Content);
