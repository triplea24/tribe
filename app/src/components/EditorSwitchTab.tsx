import React from "react";
import { Grid, Button, Theme } from "@material-ui/core";
import { connect } from "react-redux";
import { withStyles } from "@material-ui/core/styles";

import { changeEditorMode } from "../actions";

interface Props {
  classes?: any;
  editorMode: string;
  changeEditorMode: Function;
}

const tabs = [
  { key: "article", text: "Article" },
  { key: "post", text: "Post" },
  { key: "question", text: "Question", disabled: true },
  { key: "event", text: "Event", disabled: true }
];

class EditorSwitchTab extends React.Component<Props> {
  render() {
    const { editorMode, classes } = this.props;
    const Buttons = tabs.map(({ key, text, disabled = false }) => (
      <Grid
        key={key}
        item
        lg={"auto"}
        onClick={() => !disabled && this.props.changeEditorMode(key)}
      >
        <Button
          className={
            editorMode === key ? classes.activeButton : classes.inactiveButton
          }
        >
          {text}
        </Button>
      </Grid>
    ));
    return (
      <Grid container spacing={8}>
        {Buttons}
      </Grid>
    );
  }
}

interface ReduxState {
  editorMode: string;
}
const mapStateToProps = ({ editorMode = "article" }: ReduxState) => ({
  editorMode
});
const withRedux = connect(
  mapStateToProps,
  { changeEditorMode }
);

const styles = (theme: Theme) => ({
  activeButton: {
    background: "#f0f3ff",
    color: "#3e3f42",
    borderRadius: 19
  },
  inactiveButton: {
    borderRadius: 19
  }
});

const withMaterialUI = withStyles(styles);

// TODO: Should use compose
export default withRedux(withMaterialUI(EditorSwitchTab));
