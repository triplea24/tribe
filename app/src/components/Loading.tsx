import React from "react";
import {
  Theme,
  DialogContent,
  Dialog,
  CircularProgress
} from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import { connect } from "react-redux";

interface Props {
  classes?: any;
  loading: boolean;
}
class Loading extends React.Component<Props> {
  render() {
    const { classes, loading } = this.props;
    return (
      <Dialog open={loading} className={classes.root}>
        <DialogContent>
          <CircularProgress className={classes.progress} />
        </DialogContent>
      </Dialog>
    );
  }
}

const styles = (theme: Theme) => ({
  root: {
    width: "200px",
    marginLeft: "40%",
    backgroundColor: "transparent"
  },
  progress: {}
});

interface ReduxState {
  loading: boolean;
}
const mapStateToProps = ({ loading }: ReduxState) => ({
  loading
});
const withRedux = connect(
  mapStateToProps,
  {}
);
const withMaterialUI = withStyles(styles);
export default withRedux(withMaterialUI(Loading));
