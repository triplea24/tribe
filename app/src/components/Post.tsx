import React from "react";
import {
  Theme,
  Button,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  DialogActions,
  Collapse
} from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import Avatar from "@material-ui/core/Avatar";
import Typography from "@material-ui/core/Typography";
import DeleteIcon from "@material-ui/icons/Delete";
import { connect } from "react-redux";

import { removeContent } from "../actions";

const styles = (theme: Theme) => ({
  root: {
    width: "100%",
    backgroundColor: theme.palette.background.paper
  },
  inline: {
    display: "inline"
  }
});

interface Props {
  classes?: any;
  avatar: string;
  avatarAlt: string;
  type: string;
  title?: string;
  body: string;
  id: string;
  removeContent: any;
}

interface State {
  isConfirmationDialogOpen: boolean;
  visible: boolean;
}

class Content extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      isConfirmationDialogOpen: false,
      visible: false
    };
  }
  handleDelete = () => {
    this.props.removeContent(this.props.id);
    this.dismissDialog();
  };
  showDialog = () => {
    this.setState({ isConfirmationDialogOpen: true });
  };
  dismissDialog = () => {
    this.setState({ isConfirmationDialogOpen: false });
  };
  componentDidMount() {
    this.setState({ visible: true });
  }
  componentWillUnmount() {
    this.setState({ visible: false });
  }
  render() {
    const { classes } = this.props;
    return (
      <Collapse in={this.state.visible} timeout={200}>
        <ListItem alignItems="flex-start">
          <ListItemAvatar>
            <Avatar alt={this.props.avatarAlt} src={this.props.avatar} />
          </ListItemAvatar>
          <ListItemText
            primary={this.props.type === "article" && this.props.title}
            secondary={
              <Typography
                component="span"
                className={classes.inline}
                color="textPrimary"
              >
                {this.props.body}
              </Typography>
            }
          />
          <Button onClick={this.showDialog}>
            <DeleteIcon />
          </Button>
          <Dialog
            open={this.state.isConfirmationDialogOpen}
            onClose={this.dismissDialog}
            aria-labelledby="alert-dialog-title"
            aria-describedby="alert-dialog-description"
          >
            <DialogTitle id="alert-dialog-title">{"Delete"}</DialogTitle>
            <DialogContent>
              <DialogContentText id="alert-dialog-description">
                {`Are you sure you want to delete this ${this.props.type}?`}
              </DialogContentText>
            </DialogContent>
            <DialogActions>
              <Button onClick={this.dismissDialog} color="primary">
                No
              </Button>
              <Button onClick={this.handleDelete} color={"secondary"} autoFocus>
                Yes
              </Button>
            </DialogActions>
          </Dialog>
        </ListItem>
      </Collapse>
    );
  }
}

const withRedux = connect(
  null,
  { removeContent }
);
const withMaterialUI = withStyles(styles);
export const ContentComponent = withMaterialUI(Content);
export default withRedux(withMaterialUI(Content));
