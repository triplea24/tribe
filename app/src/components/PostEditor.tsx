import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { Theme } from "@material-ui/core";
import Paper from "@material-ui/core/Paper";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Avatar from "@material-ui/core/Avatar";
import Grid from "@material-ui/core/Grid";
import AttachmentIcon from "@material-ui/icons/Attachment";
import PhotoIcon from "@material-ui/icons/Photo";
import LibraryBookIcon from "@material-ui/icons/LibraryBooks";
import MoreIcon from "@material-ui/icons/MoreHoriz";

interface Props {
  classes?: any;
}

class PostEditor extends React.Component<Props> {
  render() {
    const { classes } = this.props;
    const placeholder = `What's on your mind, Soheil?`;
    return (
      <Paper className={classes.root}>
        <Grid container spacing={8}>
          <Grid item lg={12}>
            <Avatar
              src={"https://material-ui.com/static/images/avatar/1.jpg"}
            />
            <TextField
              className={classes.body}
              multiline={true}
              fullWidth={true}
              rows={5}
              placeholder={placeholder}
            />
          </Grid>
          <Grid item lg={"auto"}>
            <Button
              variant="outlined"
              color="inherit"
              className={classes.fileButton}
            >
              <PhotoIcon />
              Photo/Video
            </Button>
          </Grid>
          <Grid item lg={"auto"}>
            <Button
              variant="outlined"
              color="inherit"
              className={classes.fileButton}
            >
              <LibraryBookIcon />
              Article
            </Button>
          </Grid>
          <Grid item lg={"auto"}>
            <Button
              variant="outlined"
              color="inherit"
              className={classes.fileButton}
            >
              <AttachmentIcon />
              File
            </Button>
          </Grid>
          <Grid item lg={"auto"}>
            <Button
              variant="outlined"
              color="inherit"
              className={classes.fileButton}
            >
              <MoreIcon />
            </Button>
          </Grid>
          <Grid item lg={6} />
          <Grid item lg={1} justify={"flex-end"}>
            <Button className={classes.button}>Post</Button>
          </Grid>
        </Grid>
      </Paper>
    );
  }
}

const styles = ({ spacing }: Theme) => ({
  root: {
    flexGrow: 1,
    marginLeft: 123,
    marginRight: 163,
    padding: 18
  },
  body: {},
  button: {
    background: "#34aa44",
    color: "#ffffff",
    paddingLeft: 24,
    paddingRight: 24,
    paddignTop: 8,
    paddingBottom: 8,
    size: 14
  },
  avatar: {
    margin: 10
  }
});

export default withStyles(styles)(PostEditor);
