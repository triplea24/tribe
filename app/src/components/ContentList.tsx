import React from "react";
import { Theme, Divider } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import { connect } from "react-redux";

import Content from "./Content";
import { loadContents } from "../actions";

interface Props {
  classes?: any;
  contents: any;
  loadContents: any;
}

class ContentList extends React.Component<Props> {
  render() {
    const { classes, contents } = this.props;
    return (
      <List className={classes.root}>
        {contents &&
          Object.keys(contents)
            .reverse()
            .map(key => {
              const object = contents[key];
              const { _id, type, body, title } = object;
              return (
                <React.Fragment key={_id}>
                  <Content
                    id={_id}
                    type={type}
                    title={title}
                    body={body}
                    avatar={
                      "https://material-ui.com/static/images/avatar/1.jpg"
                    }
                    avatarAlt={"Soheil Alavi"}
                  />
                  <Divider />
                </React.Fragment>
              );
            })}
      </List>
    );
  }
  componentDidMount() {
    this.props.loadContents();
  }
}

const styles = (theme: Theme) => ({
  root: {
    width: "100%",
    backgroundColor: theme.palette.background.paper
  },
  inline: {
    display: "inline"
  }
});
const withMaterialUI = withStyles(styles);

const mapStateToProps = ({ contents }: any) => ({
  contents
});
const withRedux = connect(
  mapStateToProps,
  { loadContents }
);

export const ContentListComponent = withMaterialUI(ContentList);
export default withRedux(withMaterialUI(ContentList));
