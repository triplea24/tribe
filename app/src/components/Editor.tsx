import React from "react";
import PostEditor from "./PostEditor";
import ArticleEditor from "./ArticleEditor";
import { connect } from "react-redux";

export enum EditorType {
  Article = "Article",
  Post = "Post"
}
interface Props {
  classes?: any;
  type?: EditorType;
}
class Editor extends React.Component<Props> {
  render() {
    const { type } = this.props;
    if (type === EditorType.Post) {
      return <PostEditor />;
    }
    return <ArticleEditor />;
  }
}

interface ReduxState {
  editorMode: string;
  title?: string;
  bodyContent: string;
}

const defaultEditorMode = "article";

const mapStateToProps = ({
  editorMode = defaultEditorMode,
  title,
  bodyContent
}: ReduxState) => ({
  type: editorMode === "article" ? EditorType.Article : EditorType.Post,
  title,
  bodyContent
});
const withRedux = connect(mapStateToProps);

export default withRedux(Editor);
