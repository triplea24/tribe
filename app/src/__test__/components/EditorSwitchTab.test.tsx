import React from "react";
import renderer from "react-test-renderer";

import { EditorSwitchTabComponent } from "../../components/EditorSwitchTab";

describe("EditorSwitchTab", () => {
  it("should be rendered without crash", () => {
    const tree = renderer
      .create(
        <EditorSwitchTabComponent
          editorMode={"article"}
          changeEditorMode={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
