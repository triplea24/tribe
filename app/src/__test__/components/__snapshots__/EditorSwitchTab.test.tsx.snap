// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorSwitchTab should be rendered without crash 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-spacing-xs-8-25"
>
  <div
    className="MuiGrid-item-4 MuiGrid-grid-lg-auto-72"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root-126 MuiButton-root-100 MuiButton-text-102 MuiButton-flat-105 EditorSwitchTab-activeButton-1"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-101"
      >
        Article
      </span>
      <span
        className="MuiTouchRipple-root-129"
      />
    </button>
  </div>
  <div
    className="MuiGrid-item-4 MuiGrid-grid-lg-auto-72"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root-126 MuiButton-root-100 MuiButton-text-102 MuiButton-flat-105 EditorSwitchTab-inactiveButton-2"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-101"
      >
        Post
      </span>
      <span
        className="MuiTouchRipple-root-129"
      />
    </button>
  </div>
  <div
    className="MuiGrid-item-4 MuiGrid-grid-lg-auto-72"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root-126 MuiButton-root-100 MuiButton-text-102 MuiButton-flat-105 EditorSwitchTab-inactiveButton-2"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-101"
      >
        Question
      </span>
      <span
        className="MuiTouchRipple-root-129"
      />
    </button>
  </div>
  <div
    className="MuiGrid-item-4 MuiGrid-grid-lg-auto-72"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root-126 MuiButton-root-100 MuiButton-text-102 MuiButton-flat-105 EditorSwitchTab-inactiveButton-2"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-101"
      >
        Event
      </span>
      <span
        className="MuiTouchRipple-root-129"
      />
    </button>
  </div>
</div>
`;
